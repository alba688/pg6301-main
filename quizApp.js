import express from "express";
import {randomQuestion} from "./questions.js";

export const QuizApp = express.Router();

QuizApp.get("/random", (req, res) => {
    // res.send("Test random question");
    const { id, question, answers } = randomQuestion();
    res.json({ id, question, answers });
});

QuizApp.post("/answer", (req, res) => {
    // read body as json
    // check if answer is correct or incorrect
    // set a cookie
    // read cookie in /score

    res.end();
})