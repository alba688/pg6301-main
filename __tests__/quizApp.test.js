import express from "express";
import request from "supertest";
import {QuizApp} from "../quizApp.js";

const app = express();
app.use("/quiz", QuizApp);

describe("The quiz application", () => {
   it("returns a random question", async () => {
      const response = await request(app).get("/quiz/random").expect(200);
      expect(response.body).toMatchSnapshot({
         id: expect.any(Number),
         question: expect.any(String),
         answers: expect.any(Object)
      });
   });

   it("responds to correct answers", async () => {
      await request(app)
          .post("/quiz/answer")
          .send({ id: 101, answer: "answer_b"})
          .expect({ result: "correct"});
   })

   it("responds to incorrect answers", async () => {
      await request(app)
          .post("/quiz/answer")
          .send({ id: 101, answer: "answer_d"})
          .expect({ result: "incorrect"});
   })

});