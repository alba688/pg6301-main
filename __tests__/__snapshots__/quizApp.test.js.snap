// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The quiz application returns a random question 1`] = `
Object {
  "answers": Any<Object>,
  "id": Any<Number>,
  "question": Any<String>,
}
`;
